kalman filter for localization

TakaHoribe opened the pull request:
## Status
**DEVELOPMENT**

## Description
Pose and twist fusion for robust localization. The architecture and design follows the discussion of autowarefoundation/autoware_ai#461.

## Related Issue
autowarefoundation/autoware_ai#581 : kalman filter design
autowarefoundation/autoware_ai#461 : localization overall architecture


## Todos
- [x] Tests
- [x] Documentation


## Steps to Test or Reproduce
```
roslaunch kalman_filter_localizer kalman_filter_localizer.launch 
```

or, you can run from run_time manager.

![54423273-a190f480-4753-11e9-8fb3-68a8618ce099](https://user-images.githubusercontent.com/21360593/54441767-c059b000-4780-11e9-8c02-9e5af9305c0d.png)


## Readme
check this [readme](https://github.com/TakaHoribe/Autoware/blob/feature/kalman_filter/ros/src/computing/perception/localization/packages/kalman_filter_localizer/README.md). Functional details are described here.

kfunaoka committed:
[Release] 1.9.1 (#1691)

harderthan committed:
Feature/edit typo (#1769)

Fix language.

kfunaoka committed:
[Release] 1.10.0 (#1898)

amc-nu committed:
Update README.md

gbiggs committed:
Update contact details

isamu-takagi committed:
Feature/update autoware launcher (#2056)

* Add prototype of plugin file editor

* Change logger level

* Support command line options

* Update gazebo simulator panel

* Add plugin edit tool and update plugins

* Fix a bug of transform edit

* Fix setup.py to install all packages

* Update the code for compatibility with python3

* Update the code for colcon build

* updated plugin for lane_rule, lane_stop, and voxel_grid_filter

* Fix that the plugin_description is not installed

* Fix launch file error

* Fix test failed

* Skip test of unbuilt packages

* Add run script

* Add profile description

* Fix launch file install path of point_downsampler

* Update README file

* Update README file for plugin file

* Fix README file

* Add notice to run script

TakaHoribe commented:
I changed the branch to my fork as commented in autowarefoundation/autoware#2130

gbiggs committed:
Merge pull request #2039 from gbiggs/master

Update contact details

sgermanserrano reviewed (changes requested):
@TakaHoribe have you considered using one of the ready available libraries for the generic part of the KF as @amc-nu suggested [here](https://github.com/CPFL/Autoware/pull/2045#issuecomment-473299787)?

TakaHoribe commented:
> @TakaHoribe have you considered using one of the ready available libraries for the generic part of the KF as @amc-nu suggested [here](https://github.com/CPFL/Autoware/pull/2045#issuecomment-473299787)?

@sgermanserrano Yes, I have seen some libraries, but I couldn't judge which one should we use soon. Actually, in this kf package, I used only the most basic matrix calculation of Kalman Filter, which is just a few lines, and so I thought scratch would be better in the viewpoint of development and debugging of the first try.
At first, we should make this architecture autowarefoundation/autoware_ai#461 work. We can discuss later about which Bayes algorithm or which library is better.
Of course, I agree to use well-managed open libraries, and we can replace my kf code to other libraries in the future, if it is worth to do so.

kfunaoka committed:
Update CHANGELOGs

kfunaoka committed:
Replace URI in CHANGELOGs

kfunaoka committed:
Set version 1.10.0

kfunaoka committed:
Creanup rst format

kfunaoka committed:
1.11.0

amc-nu committed:
Change RTM default topic to match UKF node launch file (#2142)

kfunaoka committed:
[Release] 1.11.0 (#2140)

gbiggs committed:
Remove old issue and PR templates (#2148)

gbiggs committed:
Update repository info files (#2149)

* Add missing info files

* Update contributing info file

TakaHoribe committed:
kalman_filter: add all commits for signoff

Signed-off-by: TakaHoribe <horibe.takamasa@gmail.com>

TakaHoribe committed:
kalman_filter: change const to constexpr

Signed-off-by: TakaHoribe <horibe.takamasa@gmail.com>

TakaHoribe committed:
kalman_filter: modify xml format & license format

Signed-off-by: TakaHoribe <horibe.takamasa@gmail.com>

Commit:
Merge left over features from around the 1.11 release (#2151)

* Fix deceleration value bug in following state

inv_time will be negatived when forward object too close, but deceleration_critical will be smaller than max_deceleration

* fix bug of max split size

* change conversion formula from GB to Byte

* Update twist_filter.cpp (#1932)

* Feature/edit typo (#1769)

Fix language.

* Update twist_filter.cpp

* Fix/compare map filter squared distance (#1896)

* fix squared distance

* update default param

* update default param

* update launch file

* update interface.yaml

* Added AW Lexus model (#2117)

* Added AW Lexus model

* Adapt urdf to match mesh orientation

* Replace IV texture with AW (#2119)

* Update rect_class_score.h (#1825)

fixed typo for class label for OVEN - 'over' should be 'oven'

* Fixed wrong assignement of message type to package (#1706)

The message type ConfigWaypointLoader has been assigned to autoware_config_msgs in https://github.com/CPFL/Autoware/commit/8bf1c47fbc18a0303fda23c601dfe959a3afbc41.
This fixes the issue https://github.com/CPFL/Autoware/issues/1705.

* fix RuntimeManager on lxterminal #1203 (#1204)

* fix RuntimeManager on lxterminal #1203

* Modify the different behavior in gnome-terminal environment

* Set OPTION_TITLE in meta-terminal and xfce4-terminal environment

* Update run

* Add missing install for image resources on camera calibrator (#2137)

* Add missing install for image resources

* Switched to share

TakaHoribe commented:
> @TakaHoribe why did you categorize it as `fusion_localizer`?

@amc-nu Because the purpose of this kalman filter node is to integrate multiple localize information(NDT pose, GPS pose and can/IMU twist). This module doesn't generate localized position on its own. autowarefoundation/autoware_ai#461 : localization overall architecture

TakaHoribe committed:
kalman_filter: add figures for kalman filter quation

Signed-off-by: TakaHoribe <horibe.takamasa@gmail.com>

TakaHoribe committed:
kalman_filter: move kalman_filter libraries from kalman_filter_localizer pkg to amathutils_lib

Signed-off-by: TakaHoribe <horibe.takamasa@gmail.com>

Commit:
Add .repos file for setting up an Autoware workspace using vcs (#2161)

* Add .repos file for setting up an Autoware workspace using vcs

Signed-off-by: Geoffrey Biggs <geoff.biggs@tier4.jp>

* Use https protocol for git repositories

Signed-off-by: Geoffrey Biggs <geoff.biggs@tier4.jp>

* Use lgsvl_msgs from binaries

Signed-off-by: Geoffrey Biggs <geoff.biggs@tier4.jp>

* Use American English for repository names

Signed-off-by: Geoffrey Biggs <geoff.biggs@tier4.jp>

* Break spelling

Signed-off-by: Geoffrey Biggs <geoff.biggs@tier4.jp>

TakaHoribe committed:
kalman_filter: fix CMakelists for colcon build error & change initial covariance from 1e3 to 1e15

Signed-off-by: TakaHoribe <horibe.takamasa@gmail.com>

TakaHoribe committed:
kalman_filter: modify yaw calc functiuon

Signed-off-by: TakaHoribe <horibe.takamasa@gmail.com>

kfunaoka committed:
Remove enablePlannerDynamicSwitch (#1919)

* Remove enablePlannerDynamicSwitch from lane_select.launch

Signed-off-by: Kenji Funaoka <kfunaoka@gmail.com>

* Remove enablePlannerDynamicSwitch from lane_select_core.cpp

Signed-off-by: Kenji Funaoka <kfunaoka@gmail.com>

* Remove enablePlannerDynamicSwitch from velocity_set.launch

Signed-off-by: Kenji Funaoka <kfunaoka@gmail.com>

* Remove enablePlannerDynamicSwitch from velocity_set.cpp

Signed-off-by: Kenji Funaoka <kfunaoka@gmail.com>

* Remove enablePlannerDynamicSwitch from dp_planner_core.h

Signed-off-by: Kenji Funaoka <kfunaoka@gmail.com>

* Remove enablePlannerDynamicSwitch from dp_planner.launch

Signed-off-by: Kenji Funaoka <kfunaoka@gmail.com>

* Remove enablePlannerDynamicSwitch from dp_planner_core.cpp

Signed-off-by: Kenji Funaoka <kfunaoka@gmail.com>

* Remove enablePlannerDynamicSwitch from autoware_launcher

Signed-off-by: Kenji Funaoka <kfunaoka@gmail.com>

sgermanserrano committed:
Updated paths in quick_start launch files and parameters in default.rviz (#2206)

Signed-off-by: sgermanserrano <servando.german.serrano@linaro.org>

drwnz committed:
Fix bug that limits the data rate of DataRateCheckerPlugin (#2205)

* Repair bug that limits the data rate of DataRateCheckerPlugin

- A larger message buffer was added to message callbacks of the topic subscribers in DataRateCheckerPlugin
- Handling of timing was changed from std::chrono to ros::Time
- Maximum frequency selection from the Min. Hz option was increased
- Exception handling was improved to be more similar to the Python tool rostopic hz

Signed-off-by: David <david@lifecyclist.co.nz>

* Remove unused variable in header file

Signed-off-by: David <david@lifecyclist.co.nz>

* Apply ROS clang format

Signed-off-by: David <david@lifecyclist.co.nz>

* Revert "Apply ROS clang format"

This reverts commit 88563f1aa768430d78b6e4f9f9612d0680ae2385.

Signed-off-by: David <david@lifecyclist.co.nz>

snehagn committed:
Adding install directive for node (#2195)

Issue: executable is not found/built with colcon_release, added install targets.
Signed-off-by: snehagn <sganesh@autonomoustuff.com>

wep21 committed:
fix install directive for waypoint extractor (#2219)

Signed-off-by: wep21 <border_goldenmarket@yahoo.co.jp>

TakaHoribe committed:
kalman_filter: update readme

Signed-off-by: TakaHoribe <horibe.takamasa@gmail.com>

TakaHoribe committed:
ekf_localizer: change package name from kalman filter localozer

Signed-off-by: TakaHoribe <horibe.takamasa@gmail.com>

TakaHoribe committed:
twist_generator: add pkg

Signed-off-by: TakaHoribe <horibe.takamasa@gmail.com>

TakaHoribe committed:
ekf_localizer: add use_ndt_pose_with_covariance option

Signed-off-by: TakaHoribe <horibe.takamasa@gmail.com>

TakaHoribe committed:
twist_generator: modify in runtime_manager.yaml

Signed-off-by: TakaHoribe <horibe.takamasa@gmail.com>

TakaHoribe committed:
ekf_localizer: bug fix of yaw error > 2*pi & add debug messages

Signed-off-by: TakaHoribe <horibe.takamasa@gmail.com>

TakaHoribe committed:
twist_generator: add runtime_manager & change node name

Signed-off-by: TakaHoribe <horibe.takamasa@gmail.com>

TakaHoribe committed:
ekf_localizer: change params & default values

Signed-off-by: TakaHoribe <horibe.takamasa@gmail.com>

TakaHoribe committed:
Merge remote-tracking branch 'upstream/master' into feature/kalman_filter

TakaHoribe committed:
ekf_localizer: remove config file & modify readme

Signed-off-by: TakaHoribe <horibe.takamasa@gmail.com>

TakaHoribe committed:
ekf_localizer: change parameters

Signed-off-by: TakaHoribe <horibe.takamasa@gmail.com>

TakaHoribe commented:
move to autowarefoundation/autoware#2247 to fit v.1.12 release process.
